File: Shank/benchMark/MatrixMultiply/matrixMultipy.c
typedef struct row {
  long cells[586];
} row;

typedef struct matix {
  row rows[586];
} matrix;

void initMatrix(matrix *c, int seed) {
  int i, j = 0;
  for (i = 0; i < 585; i++) {
    for (j = 0; j < 585; j++) {
      c->rows[i].cells[j] = seed * (i + j);
    }
  }
}
void matrixMultiply(matrix *a, matrix *b, matrix *c) {
  int i, j, k, sum = 0;
  for (i = 0; i < 585; i++) {
    for (j = 0; j < 585; j++) {
      for (k = 0; k < 585; k++) {
        sum += a->rows[i].cells[k] * b->rows[k].cells[j];
      }
      c->rows[i].cells[j] = sum;
    }
  }
}
int main() {
  matrix a, b, c;
  initMatrix(&a, 1);
  initMatrix(&b, 3);
  matrixMultiply(&a, &b, &c);
}

File: Shank/benchMark/MatrixMultiply/matrixMultipy.py
# TODO: make matrix class
def initMatrix(seed):
    return [[seed * (i + j) for j in range(585)] for i in range(585)]
a = initMatrix(1)
b = initMatrix(3)
c = [[0 for x in range(585)] for x in range(585)]

def matrixMultiply(a,b,c):
    for i in range(585):
        for j in range(585):
            sum = 0
            for k in range(585):
                sum = sum + a[i][k] * b[k][j]
        c[i][j] = sum
matrixMultiply(a,b,c)
